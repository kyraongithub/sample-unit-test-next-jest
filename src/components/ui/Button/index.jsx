import React from 'react';

const Button = (props) => {
  const clickHandler = () => {
    if (props.onClick) {
      props.onClick();
    }
  };

  return (
    <button {...props} onClick={clickHandler}>
      {props.children}
    </button>
  );
};

export default Button;
