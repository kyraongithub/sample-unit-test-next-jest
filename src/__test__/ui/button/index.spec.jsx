import Button from '@/components/ui/Button';
import { fireEvent, render } from '@testing-library/react';

describe('Button Component', () => {
  it('should render button', () => {
    const { container } = render(<Button type="primary">hello world</Button>);
    expect(container).toMatchSnapshot();
  });

  it('should handle click button', () => {
    const { container } = render(
      <Button type="primary" onClick={() => {}}>
        hello world
      </Button>
    );
    fireEvent.click(container.firstChild);
    expect(container).toMatchSnapshot();
  });
});
