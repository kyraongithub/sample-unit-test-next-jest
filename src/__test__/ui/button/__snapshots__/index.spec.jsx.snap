// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should handle click button 1`] = `
<div>
  <button
    type="primary"
  >
    hello world
  </button>
</div>
`;

exports[`Button Component should render button 1`] = `
<div>
  <button
    type="primary"
  >
    hello world
  </button>
</div>
`;
